command/rootchain/params.go

```diff
@@ -0,0 +1,203 @@
+package rootchain
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/command/helper"
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/umbracle/ethgo/abi"
+)
+
+const (
+	dirFlag           = "dir"
+	rootchainAddrFlag = "rootchain-addr"
+	eventABIFlag      = "event-abi"
+	methodABIFlag     = "method-abi"
+	localAddrFlag     = "local-addr"
+	payloadTypeFlag   = "payload-type"
+)
+
+var (
+	params = &rootchainParams{}
+)
+
+const (
+	defaultRootchainName = "rootchain.json"
+)
+
+var (
+	errInvalidLocalAddr   = errors.New("invalid local SC address")
+	errInvalidPayloadType = errors.New("invalid payload type")
+	errInvalidEventABI    = errors.New("invalid event ABI")
+	errInvalidMethodABI   = errors.New("invalid method ABI")
+	errUnableToReadConfig = errors.New("unable to read rootchain config")
+)
+
+type rootchainParams struct {
+	configPath    string
+	rootchainAddr string
+	eventABI      string
+	methodABI     string
+	localAddr     string
+	payloadType   uint64
+
+	config *rootchain.Config
+}
+
+func (p *rootchainParams) getRequiredFlags() []string {
+	return []string{
+		rootchainAddrFlag,
+		eventABIFlag,
+		methodABIFlag,
+		localAddrFlag,
+	}
+}
+
+func (p *rootchainParams) validateFlags() error {
+	// Check if the payload type is supported
+	if err := validatePayloadType(p.payloadType); err != nil {
+		return err
+	}
+
+	// Check if the event ABI is valid
+	if err := p.validateEventABI(); err != nil {
+		return err
+	}
+
+	// Check if the method ABI is valid
+	if err := p.validateMethodABI(); err != nil {
+		return err
+	}
+
+	// Check if the local address is valid
+	return p.validateLocalAddress()
+}
+
+func validatePayloadType(payloadType uint64) error {
+	if rootchain.PayloadType(payloadType) != rootchain.ValidatorSetPayload {
+		return errInvalidPayloadType
+	}
+
+	return nil
+}
+
+func (p *rootchainParams) validateEventABI() error {
+	_, err := abi.NewEvent(p.eventABI)
+
+	if err != nil {
+		return errInvalidEventABI
+	}
+
+	return nil
+}
+
+func (p *rootchainParams) validateMethodABI() error {
+	_, err := abi.NewABI(p.methodABI)
+	if err != nil {
+		return errInvalidMethodABI
+	}
+
+	return nil
+}
+
+func (p *rootchainParams) validateLocalAddress() error {
```

Comment:
Do we need to validate that local-addr and rootchain-addr are of type.Address, that the format is good?

Reply:
The rootchain address is not of type `types.Address`, but a WS endpoint.
The local-addr value is of type `types.Address`.

I don't see how we can check the validity of the local address, apart from just making sure it's set ("0x1" is a valid address that will get expanded for example)

Reply:
kk :D